packages/components/src/search-control/types.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import type { KeyboardEventHandler } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import type { WPElement } from '@wordpress/element';
+
+export type SearchControlProps = {
+	className: string | undefined;
+	onChange: ( arg0: string ) => {};
+	onKeyDown: KeyboardEventHandler< HTMLInputElement >;
+	value: string | number;
+	label: string;
+	placeholder: string | undefined;
+	hideLabelFromVision: boolean | undefined;
+	help: string | WPElement | undefined;
```

Comment:
Instead of using `undefined`, you can make those props optional.

```suggestion
	className?: string;
	onChange: ( arg0: string ) => {};
	onKeyDown: KeyboardEventHandler< HTMLInputElement >;
	value: string | number;
	label: string;
	placeholder?: string;
	hideLabelFromVision?: boolean;
	help?: string | WPElement;
```